src/Parser/ASTConverter.cs

```diff
@@ -955,6 +955,16 @@ void VisitDeclaration(Declaration decl, AST.Declaration _decl)
             _decl.OriginalPtr = originalPtr;
 
             NativeObjects.Add(decl);
+
+            for (uint i = 0; i < decl.RedeclarationsCount; i++)
+            {
+                var redecl = decl.GetRedeclarations(i);
+                if (redecl.OriginalPtr != _decl.OriginalPtr)
```

Comment:
This isn't necessary because the current declaration is already added to the cache.